### ISSUE TITLE
so100 demo

### ISSUE THREAD

**Issue by richardnsw** [MINT-SJTU/Evo-1#9]
Great paper and VLA. The published results are impressive
I am just trying the SO100 inference on my robot just to see how it goes. 
Setting up was very easy and thanks for the detailed instructions. 
I have the robot on the table and a camera to the side. the robot is moving around a lot but not performing the task.
 Is the example checkpoint and pretrained weights good to go? I noticed it said just 500 steps. could you release a more trained checkpoint just to act as a starting point for a demo, or am I doing something wrong? 
Thanks again - great paper/code

**Comment by DorayakiLin** [comment]
Hello! Does your SO100 robotic arm vibrate violently? If so, please check if flash-atten is installed correctly. 
May I ask how many demostrations you collect for this task? And can you provide more details of your training?
I will update a more trained checkpoint and demo videos for so100 and so101 in these days.

**Comment by DorayakiLin** [comment]
Hello!
Was the model successfully stored at 2500? Because I saw in the instructions that it should be saved every 2500 steps.

**Comment by DorayakiLin** [comment]
Hello!
Are stage1 and stage2 using the same device?
And please verify that the checkpoint folder contains the following files: bf16_zero_pp_rank_0_mp_rank_00_optim_states.pt, checkpoint.json, config.json, mp_rank_00_model_states.pt, norm_stats.json

<img width="479" height="138" alt="Image" src="https://github.com/user-attachments/assets/e393543c-9183-48c4-b05f-d71fad37cbf3" />

**Comment by jcl2023** [comment]
I am able to run stage2 training now. Will let you know how it goes.

>>If you’re willing, you may consider joining our discussion group.
Yes, I like to join


Thanks,

**Comment by jcl2023** [comment]
I am able to training stage2 50k successfully
stage is trained with 2500 steps

WIth this, the inference is quite bad. You can see the video below.
Also attach 2 json files mentioned in the sec 5.4

inference script:
lerobot-record \
   --robot.type=so101_follower \
   --robot.port=/dev/ttyACM1 \
   --robot.id=my_awesome_follower_arm_yellow \
   --robot.cameras="{
     top: {type: intelrealsense, serial_number_or_name: 927522070838, width: 640, height: 480, fps: 30, color_mode: RGB, use_depth: false},
     front: {type: opencv, index_or_path: 8, width: 640, height: 480, fps: 30}
   }" \
   --display_data=true \
   --teleop.type=so101_leader \
   --teleop.port=/dev/ttyACM0 \
   --teleop.id=my_awesome_leader_arm_yellow \
   --dataset.episode_time_s=60 \
   --dataset.reset_time_s=10 \
   --dataset.num_episodes=5 \
   --dataset.repo_id=${HF_USER}/eval_evo1_50k_place_green_block \
   --dataset.single_task="place green blocks into bin" \
   --policy.path=/home/leon/.cache/huggingface/hub/models--pmcheng--evo1_50k/

Do you know what could be the issues for such bad performance?

[model_config.json](https://github.com/user-attachments/files/23694177/model_config.json)
[config.json](https://github.com/user-attachments/files/23694178/config.json)

https://github.com/user-attachments/assets/d16461d4-8968-4b9b-94c1-6cdb7b3e2e2b

**Comment by DorayakiLin** [comment]
We collect 100 demonstration and used the following training configuration on so100/101:
one 1*A100 ,stage1 batchsize16 5ksteps, stage2 batchsize16 30ksteps
The final execution should be relatively smooth.

**Comment by jcl2023** [comment]
>>3.When I check the [model_config.json](https://github.com/user-attachments/files/23694177/model_config.json) and [config.json](https://github.com/user-attachments/files/23694178/config.json), I've noticed that this is the sample I provided, not the one you modified according to your own situation. For example, the camera name in the config.json does not correspond with the camera you are actually using(observation.images.horizon/observation.images.front).

Sorry, I posted a wrong model_config.json. The correct one is below:

{
  "device": "cuda",
  "run_name": "Evo1_metaworld_stage2",
  "vlm_name": "OpenGVLab/InternVL3-1B",
  "action_head": "flowmatching",
  "return_cls_only": false,
  "disable_wandb": true,
  "dataset_type": "lerobot",
  "data_paths": null,
  "dataset_config_path": "dataset/config.yaml",
  "image_size": 448,
  "binarize_gripper": false,
  "use_augmentation": true,
  "lr": 1e-05,
  "batch_size": 4,
  "max_steps": 80000,
  "warmup_steps": 1000,
  "grad_clip_norm": 1.0,
  "weight_decay": 0.001,
  "log_interval": 10,
  "ckpt_interval": 2500,
  "save_dir": "./checkpoints/stage2",
  "resume": true,
  "resume_path": "./checkpoints/stage1/step_2500",
  "resume_pretrain": true,
  "finetune_vlm": true,
  "finetune_action_head": true,
  "per_action_dim": 24,
  "state_dim": 24,
  "horizon": 50,
  "num_layers": 8,
  "num_workers": 4,
  "dropout": 0.2,
  "action_dim": 1200
}

For config.json, just copy yours and change camera values below 
   "observation.images.top": {
      "type": "VISUAL",
      "shape": [3, 480, 640]
    },
    "observation.images.horizon": {
      "type": "VISUAL",
      "shape": [3, 480, 640]
    }

Is there anything wrong with config.json?

**Comment by jcl2023** [comment]
See config.json and model_config.json in the attachment which we used in the  latest test. so-arm101 performs like the previous video in the jira. Below is our inference script:

lerobot-record \
 --robot.type=so101_follower \
 --robot.port=/dev/ttyACM1 \
 --robot.id=my_awesome_follower_arm_yellow \
 --robot.cameras="{
 top: {type: intelrealsense, serial_number_or_name: 927522070838,
width: 640, height: 480, fps: 30, color_mode: RGB, use_depth: false},
 front: {type: opencv, index_or_path: 8, width: 640, height: 480, fps:
30}
 }" \
 --display_data=true \
 --teleop.type=so101_leader \
 --teleop.port=/dev/ttyACM0 \
 --teleop.id=my_awesome_leader_arm_yellow \
 --dataset.episode_time_s=60 \
 --dataset.reset_time_s=10 \
 --dataset.num_episodes=5 \
 --dataset.repo_id=${HF_USER}/eval_evo1_50k_place_green_block_3

[config.json](https://github.com/user-attachments/files/23736615/config.json)
[model_config.json](https://github.com/user-attachments/files/23736616/model_config.json)


Can you help us to check again what could be the root-cause?

**Comment by DorayakiLin** [comment]
1.The model_config.json you provided is from metaworld checkpoints, not the so101 checkpoint.
2.We do not use the depth information in our model. We only use the rgb information from realsense. And when deploying in so101/so100, we use hikvision camera.

**Comment by DorayakiLin** [comment]
Hello!
1.When you run the inference, is there a message saying "FlashAttention2 is not installed."?
2.When training for the stage 2, we use batchsize 16 and train for 30k steps to get the results. You can turn on the gradient_accumulation_steps and extend your training steps.

**Comment by yinxinyuChen** [comment]
> One more question,
> 
> Did you test --dataset.single_task? Does it have an effect? For ex, if the table has only red color block and --dataset.single_task "grab a green block and put it in a tray bin", will robot not do anything?

We haven’t tested this yet，but we will consider it in our future work.
